packages/xstate-wallet/src/store/state-utils.ts

```diff
@@ -70,9 +78,46 @@ export function statesEqual(left: State, right: State) {
   return hashState(left) === hashState(right);
 }
 
+function simpleAllocationsEqual(left: SimpleAllocation, right: SimpleAllocation) {
+  if (left.assetHolderAddress === right.assetHolderAddress) {
+    if ((left.allocationItems.length = right.allocationItems.length)) {
+      return left.allocationItems
+        .map(
+          (value, index, array) =>
+            value.destination === right.allocationItems[index].destination &&
+            value.amount.eq(right.allocationItems[index].amount)
+        )
+        .reduce((a, b) => a && b);
+    }
+  }
+  return false;
```

Comment:
I think this is more understandable as a sequence of conditionals joined with `&&`
```suggestion
  return (
    left.assetHolderAddress === right.assetHolderAddress &&
    left.allocationItems.length === right.allocationItems.length &&
    _.every(
      left.allocationItems,
      (value, index) =>
        value.destination === right.allocationItems[index].destination &&
        value.amount.eq(right.allocationItems[index].amount)
    )
  );
```

:caution: ^ Needs a Lodash import, can be replaced with `.map(...).reduce(...)` as above.